bluesky/run_engine.py

```diff
@@ -807,12 +810,18 @@ def _open_run(self, msg):
 
     @asyncio.coroutine
     def _close_run(self, msg):
+        if not self._run_is_open:
+            raise IllegalMessageSequence("A 'close_run' message was received "
+                                         "but there is no run open. If this "
+                                         "occurred after a pause/resume, add "
+                                         "a 'checkpoint' message after the "
+                                         "'close_run' message.")
```

Comment:
...but I did just add a checkpoint message at the very end of `PlanBase`, so our built-in plans will address the common case.